st2actions/st2actions/runners/actionchainrunner.py

```diff
@@ -142,41 +144,66 @@ def pre_run(self):
 
     def run(self, action_parameters):
         action_node = self.chain_holder.get_next_node()
-        results = {}
+
+        result = {'tasks': []}  # holds final result we store
+        context_result = {}  # holds result which is used for the template context purposes
         fail = True
+
+        error = None
+
         while action_node:
             actionexec = None
             fail = False
+            created_at = datetime.datetime.now()
+
             try:
-                resolved_params = ActionChainRunner._resolve_params(action_node, action_parameters,
-                                                                    results, self.chain_holder.vars)
-                actionexec = ActionChainRunner._run_action(action_node.ref,
-                                                           self.action_execution_id,
-                                                           resolved_params)
-            except:
-                LOG.exception('Failure in running action %s.', action_node.name)
+                resolved_params = ActionChainRunner._resolve_params(
+                    action_node=action_node, original_parameters=action_parameters,
+                    results=context_result, chain_vars=self.chain_holder.vars)
+                actionexec = ActionChainRunner._run_action(
+                    action_node=action_node, parent_execution_id=self.action_execution_id,
+                    params=resolved_params)
+            except Exception:
                 # Save the traceback and error message.
-                results[action_node.name] = {'error': traceback.format_exc(10)}
+                LOG.exception('Failure in running action %s.', action_node.name)
+                error = traceback.format_exc(10)
+                context_result[action_node.name] = {'error': error}
             else:
-                # Append full result under the node_name
-                results[action_node.name] = actionexec.result
+                # Update context result
+                context_result[action_node.name] = actionexec.result
+
+                # Render and publish variables
                 rendered_publish_vars = ActionChainRunner._render_publish_vars(
-                    action_node, actionexec.result, results, self.chain_holder.vars)
+                    action_node=action_node, execution_result=actionexec.result,
+                    previous_execution_results=context_result, chain_vars=self.chain_holder.vars)
+
                 if rendered_publish_vars:
                     self.chain_holder.vars.update(rendered_publish_vars)
             finally:
+                # Record result and resolve a next node based on the task success or failure
+                updated_at = datetime.datetime.now()
+
+                format_kwargs = {'action_node': action_node, 'action_exec_db': actionexec,
```

Comment:
If this comes from the except block action_exec_db will be None. I will fix it on the rename branch.